python/torch_mlir/__init__.py

```diff
@@ -333,105 +349,125 @@ def compile(model: torch.nn.Module,
     # backend. This separation should be visible at the Python API level, and
     # we can implement a deliberately simplified API like `torch_mlir.compile`
     # on top of those building blocks.
-    if isinstance(model, torch.jit.ScriptModule):
-        # If the user already converted the model to JIT IR themselves, just
-        # do some basic error checking, but take the model as-is.
-        for method_name in example_args._get_methods():
-            if not hasattr(model, method_name):
-                raise Exception(
-                    f"Model does not have exported method '{method_name}', "
-                    f"requested in `example_args`. Consider adding "
-                    f"`@torch.jit.export` to the method definition.")
-        scripted = model
-    elif use_tracing:
-        scripted = torch.jit.trace_module(
-            model,
-            example_args._get_for_tracing(use_tracing, ignore_traced_shapes)
+    global mlir_module
+    if not isinstance(model, torch.fx.GraphModule):
```

Comment:
we should not need this branch. We need a flag like use_tracing, except it will be `use_dynamo_export` or something like that (not sure if dynamo.export has been promoted to a more real/supported API yet. may need an "experimental" in the name somewhere). Dynamo can trace the through any torch.nn.Module, not just FX modules.